taichi/lang_util.h

```diff
@@ -199,6 +199,38 @@ class TypedConstant {
   TypedConstant(float64 x) : dt(DataType::f64), val_f64(x) {
   }
 
+  template <typename T>
+  TypedConstant(DataType dt, const T &value) : dt(dt) {
+    if (dt == DataType::f32) {
```

Comment:
Maybe we can use a `PER_TYPE` here? https://github.com/taichi-dev/taichi/blob/43e1da5d511f6ac949504c80fa242b14fde1cf12/taichi/lang_util.h#L23

Reply:
But we don't have `val_gen` and `val_unknown`...